curvesim/templates/data_source.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+from os import extsep
+from os.path import join
+
+from pandas import DataFrame, read_csv
+
+from .sim_asset import SimAsset
+from .time_sequence import TimeSequence
+
+
+class DataSource(ABC):
+    @abstractmethod
+    def query(self, sim_asset: SimAsset, time_sequence: TimeSequence) -> DataFrame:
+        raise NotImplementedError
+
+
+class ApiDataSource(DataSource):
+    """WIP -- anything needed here?"""
+
+
+class FileDataSource(DataSource):
```

Comment:
Probably better to make this abstract and have children like `CsvDataSource`.  This will allow more specific logic without creating a large class with different control flows.  Later, another class ("factory") could be used to delegate to the right class.

Reply:
Properties can be `path`, `file_extension`, etc.